src/v2/ion/transformer.js

```diff
@@ -0,0 +1,185 @@
+import { _ } from 'okta';
+
+/**
+ * Transform the ion spec response into canonical format.
+ */
+
+ /**
+  * Authn V2 response
+  * @typedef {Object} AuthnResponse
+  */
+
+/**
+ * @typedef {Object} AuthnResult
+ * @property {CurrentState} currentState
+ * @property {Object=} user
+ * @property {Object=} factor
+ */
+
+/**
+ * @typedef {Object} CurrentState
+ * @property {string} stateHandle
+ * @property {string} step
+ * @property {string} intent
+ * @property {function} cancel
+ * @property {function} context
+ * @property {Remediation} remediation
+ */
+
+/**
+ * @typedef {Object} Remediation
+ * @property {RemediationObject[]} values
+ */
+
+/**
+ * @typedef {Object} RemediationObject
+ * @property {string} name
+ * @property {value[]=} value mandatory visible parameters
+ */
+
+const isObject = (x) => _.isObject(x);
+const isArray = (x) => Array.isArray(x);
+const containsAny = (xs, ys) => _.intersection(xs, ys).length > 0;
+
+const getRelObjectByName = (resp, relName) => {
+  const getRelBy_ = (result, resp_) => {
+    if (isObject(resp_)) {
+      if (resp_.rel && containsAny(resp_.rel, relName)) {
+        result.push(resp_);
+      } else {
+        Object.keys(resp_)
+          .forEach(k => {
+            getRelBy_(result, resp_[k]);
+          });
+      }
+    } else if (isArray(resp_)) {
+      resp_.forEach(x => getRelBy_(result, x));
+    }
+    return result;
+  };
+
+  return getRelBy_([], resp);
+};
+
+/**
+ *
+ * @returns {Object} actionObject
+ * @returns {string} actionObject.name
+ * @returns {function} actionObject.createRequest
+ */
+const createActionObj = (relObj) => {
+  const data = (relObj.value || [])
+    .filter(v => v.visible === false)
+    .map(v => { return { [v.name]: v.value }; })
+    .reduce((init, x) => Object.assign({}, init, x), {});
```

Comment:
I think you can simplify to `.reduce((init, x) => Object.assign(init, x), {});` (might also be faster)

Reply:
I kind of prefer not to override parameters (first one) unless it becomes issue or we have to.